### PR TITLE
Update org owners, define authority and responsibilities

### PR DESCRIPTION
**Pull request by infinisil**
After discussions with the Steering Committee to set the general direction for this, here's a hashed-out proposal by me to update org owners and define their authority and responsibilities. 

Needs to be approved by @NixOS/steering before it can be merged. Note that the convention for this repo is that the person merging also implements the changes.

Also note that the first commit is shared with #36.

### CODE REVIEW

File: doc/github-org-owners.md
Comment (winterqt) [inline doc/github-org-owners.md:35]: Could you expand on why you think this requires two+ approvals, but maintaining random unkempt repositories doesn't? (I'm not opposed, just trying to see if there's something I missed.)

I guess an app could be dangerous?

Comment (infinisil) [reply doc/github-org-owners.md:35]: The intent is that owners would only do maintenance with _low impact_ by themselves. It's nested under the "low-impact changes" point to imply that, but maybe that's not clear enough.

However, it's hard to estimate impact, so maybe err on the safe side and have this require 2+ approvals. Especially also because if necessary, 2+ org owners can delegate one of their own to have commit access to a specific repo and therefore empower them to maintain on their own.  

I'll move it for now, I convinced myself :)

**Review by Gabriella439 — APPROVED** [review]

**Review by winterqt — APPROVED** [review]

**Review by tomberek — APPROVED** [review]

File: .github/CODEOWNERS
Comment (zimbatm) [inline .github/CODEOWNERS]: How about collapsing this and the @NixOS/org team into one?

File: doc/github-org-owners.md
Comment (zimbatm) [inline doc/github-org-owners.md]:
```suggestion
The following people have the GitHub "owners" permissions on the NixOS organization:
```
Try to distinguish between real ownership and the github permissions. I expect this team to have some autonomy to unblock simple things, but overall follow the decisions of the steering comittee.

File: doc/github-org-owners.md
Comment (zimbatm) [inline doc/github-org-owners.md:14]: Maybe something for later:

I'd like us to standardize using GitHub issues to discuss GitHub-related topics. Spreading information in two places is not optimal for transparency reasons.

File: doc/github-org-owners.md
Comment (zimbatm) [inline doc/github-org-owners.md]:
```suggestion
- Receive and process requests.
  - We expect org owners to be subscribed to the NixOS/org repo.
  - Each request should land in at least 2 org owners inboxes of some kind.
```

File: doc/github-org-owners.md
Comment (zimbatm) [inline doc/github-org-owners.md]:
```suggestion
## How to contact the team
```

File: doc/github-org-owners.md
Comment (zimbatm) [inline doc/github-org-owners.md:18]: This section sounds procedural. It might make sense to add a more fuzzy higher-level description to make this less rigid.

Something like:
```suggestion
This team's role is to manage and unblock users of the github.com/NixOS GitHub organization. The @NixOS/steering gives them autonomy to handle small day-to-day tasks and expects them to escalate bigger decisions.
```

**Review by zimbatm — COMMENTED** [review]
A few ideas, but looks good overall

File: doc/github-org-owners.md
Comment (infinisil) [reply doc/github-org-owners.md]: In the future I hope we can have an approach of automatically assigning requests to different org owners, to reduce the amount of time and attention being used, but for now this seems fine :)

**Review by fpletz — APPROVED** [review]

**Comment by infinisil** [comment]
Applied @zimbatm's suggestions and included some follow-up changes.

Here's a more easily viewable diff since y'all approved the last time: https://github.com/NixOS/org/compare/6f99f49dc3a019a67eccf7f1d804ef4b50686edd..40b246011b6c71dfa8fb16eeb72ac68dc22b0986 (this diff also includes #36 due to the rebase force push though)

**Review by zimbatm — APPROVED** [review]
thanks @infinisil

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
so we have 4 SC approvals (a majority) and the proposal was created 4 days ago. If I interpret the SCs voting rules correctly that means I can implement those changes and merge this PR?

**Comment by Gabriella439** [comment]
@Lassulus: I'm going to say: yes.

I generally want us to err on the side of doing things.  If we make a mistake in interpreting the rules we can always revert it.

File: doc/github-org-owners.md
Comment (jtojnar) [inline doc/github-org-owners.md:25]: How do you imagine this to be used? In Nixpkgs, this is handled by [commiter delegation team](https://github.com/NixOS/org/blob/main/doc/github.md#nixpkgs), and in other repos like Nix, I would expect them to have owners who would ideally self-organize.

File: doc/github-org-owners.md
Comment (jtojnar) [inline doc/github-org-owners.md]: Maybe make it clearer if the 2 includes the implementer.

```suggestion
With approval by at least 2 other org owners, they can take care of implementing
```

or

```suggestion
With approval by at least one other org owner, they can take care of implementing
```

File: doc/github-org-owners.md
Comment (infinisil) [reply doc/github-org-owners.md:25]: This is just for the issue of people missing the automated org invite they get after adding themselves to `lib.maintainers`, see https://github.com/NixOS/nixpkgs/issues/234293. An org owner currently needs to resend these missed invites, since there's no automation for that. These invites give no commit access, they're just to allow requesting reviews from maintainers.

File: doc/github-org-owners.md
Comment (infinisil) [reply doc/github-org-owners.md]: The latter, I think this should be even more clear:

```suggestion
Org owners need approval from at least one other org owner to take care of implementing
```

**Comment by Lassulus** [comment]
Alright, I created a team and made everyone individually an org owner.

**Comment by Lassulus** [comment]
@NixOS/org-owners is the team btw :)

**Comment by infinisil** [comment]
@Lassulus No need for a new team, the docs you just merged document @NixOS/org!

**Comment by Lassulus** [comment]
ah well, feel free to remove that team :D

**Pull request merged**
